km/load_elf.c

```diff
@@ -270,50 +282,27 @@ km_open_elf_file(const char *path)
    }
 
    // Read ELF header
-   elf->ehdr = (Elf64_Ehdr *) malloc(sizeof(Elf64_Ehdr));
-   int nread = fread(elf->ehdr, 1, sizeof(Elf64_Ehdr), elf->file);
+   int nread = fread(&elf->ehdr, 1, sizeof(Elf64_Ehdr), elf->file);
```

Comment:
2and 3rd params should be swapped.
Current code will work, but is confusing.

Reply:
Actually this is a conscious choice since I want the return value to be the number of bytes read, not the number of records (see the man page). fread/fwrite are kind of weird since the return value is the number of 'items' transferred. I've always found it easier to make the item size '1' and deal in bytes just like read/write. I guess it's just laziness on my part.